src/Corax/Queries/MultiUnaryMatch.cs

```diff
@@ -0,0 +1,596 @@
+                                            goto NotMatch;
+                                    }
+                                }
+                            }
+                        }
+
+                        break;
+                    case IndexEntryFieldType.Raw:
+                    case IndexEntryFieldType.RawList:
+                    case IndexEntryFieldType.Invalid:
+                        break;
+                    default:
+                        if (reader.Read(comparer.FieldId, out var value) == false)
+                            goto NotMatch;
+
+                        using (_searcher.ApplyAnalyzer(value, comparer.FieldId, out var analyzedTerm))
+                        {
+                            if (comparer.Type != MultiUnaryItem.DataType.Slice || comparer.CompareLiteral(analyzedTerm) == false)
+                                goto NotMatch;
+                        }
+
+                        break;
+                }
+            }
+
+            NotMatch:
+            if (comparerId == _comparers.Length)
+            {
+                matches[currentIdx++] = matches[i];
+                // Its a match
+            }
+        }
+
+        return currentIdx;
+
+        bool IsAcceptedItem(MultiUnaryItem comparer, ReadOnlySpan<byte> sequence, in long longValue, in double doubleValue) => comparer.Type switch
+        {
+            MultiUnaryItem.DataType.Slice => comparer.CompareLiteral(sequence),
+            MultiUnaryItem.DataType.Long => comparer.CompareNumerical(longValue),
+            _ => comparer.CompareNumerical(doubleValue)
+        };
+
+        bool IsAcceptedForIterator(MultiUnaryItem comparer, in IndexEntryFieldIterator iterator) => comparer.Type switch
+        {
+            MultiUnaryItem.DataType.Slice => comparer.CompareLiteral(iterator.Sequence),
+            MultiUnaryItem.DataType.Long => comparer.CompareNumerical(iterator.Long),
+            _ => comparer.CompareNumerical(iterator.Double)
+        };
+    }
+
+    public unsafe int AndWith(Span<long> buffer, int matches)
+    {
+        using var _ = _searcher.Allocator.Allocate(sizeof(long) * buffer.Length, out Span<byte> bufferHolder);
+        var innerBuffer = MemoryMarshal.Cast<byte, long>(bufferHolder).Slice(0, buffer.Length);
+        Debug.Assert(innerBuffer.Length == buffer.Length);
+
+        var count = Fill(innerBuffer);
+        int result = 0;
+        if (count > 0)
+        {
+            fixed (long* matchesPtr = buffer)
+            fixed (long* baseMatchesPtr = innerBuffer)
+            {
+                result = MergeHelper.And(matchesPtr, buffer.Length, matchesPtr, matches, baseMatchesPtr, count);
+            }
+        }
+        
+        return result;
+    }
+
+    public void Score(Span<long> matches, Span<float> scores)
+    {
+        //TODO Federico: Can you handle that thing? I'm not sure how internally boosting flow works.
+        throw new NotImplementedException();
+    }
```

Comment:
@maciejaszyk Totally forgot about this. Let's setup a call for tomorrow and we can write the code in there.

Reply:
done